unikraft/elfloader/options.go

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2022, Unikraft GmbH and The KraftKit Authors.
+// Licensed under the BSD-3-Clause License (the "License").
+// You may not use this file except in compliance with the License.
+package elfloader
+
+import (
+	"kraftkit.sh/unikraft/app"
+)
+
+type ELFLoaderOption func(*ELFLoader) error
+
+func WithApplicationOptions(aopts ...app.ApplicationOption) ELFLoaderOption {
+	return func(ef *ELFLoader) error {
+		//ap := ef.application
+
+		//for _, o := range aopts {
+			//if err := o(&ap); err != nil {
```

Comment:
This fails right now, since `ap` is of type `app.Application` but the `app.ApplicationOption` expects an `app.application` argument.

Reply:
I would save these options into the `ELFLoader` struct and then iterate over them in the EFLLoader's constructor, e.g.

```go
type ELFLoader struct {
  [...]
  aopts []app.ApplicationOption
  [...]
}

[...]

app, err := app.NewApplicationFromOptions(eopts.aopts...)
if err != nil {
[...]
```